### DISCUSSION TITLE
Tab list as a dropdown menu

### DISCUSSION THREAD

**Discussion by fbck** [Chatterino/chatterino2#5030] [Ideas]
Having a lot of tabs becomes unwieldly, but current options for tabs, like 'toggle visibility/hide all tabs' and 'show only live tabs' still aren't all that great for making tabs visible and having a grasp on what tabs you have, and having ability to quickly access those tabs.
A tab menu - a dropdown list of tabs, that'd let you switch between tabs, that'd show up by clicking on a button or as a right-click submenu, would help keep the whole list of channels/tabs out of the way, while still having it be wholly visible and readily accessible. (and without a guesswork of 'if I switch with ctrl-tab or scroll, which order are my tabs in? how do I get to the one I want', or switching between tab visibility modes.)
Something like the tab dropdowns / tab search menus in some browsers, like this one in chrome https://static1.howtogeekimages.com/wordpress/wp-content/uploads/2020/12/google-chrome-search-tabs-hero.png

**Comment by Nerixyz** [comment]
I agree, there should be more options of displaying tabs.

`Notebook.cpp` is the class managing the layout of the tabs. Its layout is [quite involved](https://github.com/Chatterino/chatterino2/blob/master/src/widgets/Notebook.cpp#L648-L1087), which might be one reason for the lack of innovation there (mainly due to supporting multiple locations). This would need to be refactored first.

---

Linking to https://github.com/Chatterino/chatterino2/discussions/5031 for completeness. You might consider merging both into one thread.
To show images inline in Markdown, you can use `![alt text](https://image-link)` (or use `<img src="https://image-link">` where you can specify a `width`).
